plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -789,56 +793,40 @@ protected Optional<ConnectorMaterializedViewDefinition> doGetMaterializedView(Co
-        IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
-                this,
-                session,
-                storageTableName.getSchemaName(),
-                storageTableName.getTableName(),
-                Optional.empty(),
-                Optional.empty());
-        try {
-            TableMetadata metadata = getMaterializedViewTableMetadata(session, storageTableName, materializedView);
-            operations.initializeFromMetadata(metadata);
-            Table icebergTable = new BaseTable(operations, quotedTableName(storageTableName), TRINO_METRICS_REPORTER);
+        return Optional.of(getMaterializedViewDefinition(
+                materializedView.getOwner(),
+                materializedView.getViewOriginalText()
+                        .orElseThrow(() -> new TrinoException(HIVE_INVALID_METADATA, "No view original text: " + viewName)),
+                storageTableName));
+    }
 
-            return Optional.of(getMaterializedViewDefinition(
-                    icebergTable,
-                    materializedView.getOwner(),
-                    materializedView.getViewOriginalText()
-                            .orElseThrow(() -> new TrinoException(HIVE_INVALID_METADATA, "No view original text: " + viewName)),
-                    storageTableName));
+    @Override
+    public Map<String, Object> getMaterializedViewProperties(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition)
+    {
+        if (definition.getStorageTable().isEmpty()) {
+            throw new TrinoException(ICEBERG_INVALID_METADATA, "Materialized view definition is missing a storage table");
+        }
+
+        SchemaTableName storageTableName = definition.getStorageTable().get().getSchemaTableName();
+        try {
+            Table storageTable = loadTable(session, definition.getStorageTable().orElseThrow().getSchemaTableName());
+            return ImmutableMap.<String, Object>builder()
+                    .putAll(getIcebergTableProperties(storageTable))
+                    .put(STORAGE_SCHEMA, storageTableName.getSchemaName())
+                    .buildOrThrow();
         }
         catch (RuntimeException e) {
-            // The materialized view could be removed concurrently. This may manifest in a number of ways, e.g.
-            // - org.apache.iceberg.exceptions.NotFoundException when accessing manifest file
-            // - other failures when reading storage table's metadata files
-            // Retry, as we're catching broadly.
             metastore.invalidateTable(viewName.getSchemaName(), viewName.getTableName());
-            throw new MaterializedViewMayBeBeingRemovedException(e);
+            metastore.invalidateTable(storageTableName.getSchemaName(), storageTableName.getTableName());
```

Comment:
Those invalidations seem to have been moved from doGetMaterializedView.
That method was run with a retry, this one is not.

For now, i think we can do without retries (in this method we cannot "change our mind" and say the MV has been deleted) and that means we don't need invalidation anymore.